integrations/integrations.go

```diff
@@ -0,0 +1,231 @@
+	wrappedApp := &WrappedApplication{app}
+	_, err := startMonomerNode(wrappedApp, env, svrCtx)
+	if err != nil {
+		return fmt.Errorf("start Monomer node: %v", err)
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, *clientCtx, ctx, g); err != nil {
+			return fmt.Errorf("run post setup: %v", err)
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node. The
+// `MonomerGenesisPath` flag must be set in viper before invoking this.
+func startMonomerNode(wrappedApp *WrappedApplication, env *environment.Env, svrCtx *server.Context) (*node.Node, error) {
+	engineWS, err := net.Listen("tcp", viper.GetString(monomerEngineWSFlag))
+	if err != nil {
+		return nil, fmt.Errorf("create Engine websocket listener: %v", err)
+	}
+	env.DeferErr("close engine ws", engineWS.Close)
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+	cometListener, err := net.Listen("tcp", cmtListenAddr)
+	if err != nil {
+		return nil, fmt.Errorf("create CometBFT listener: %v", err)
+	}
+	env.DeferErr("close comet listener", cometListener.Close)
+
+	appdb, err := cometconfig.DefaultDBProvider(&cometconfig.DBContext{
+		ID:     "app",
+		Config: svrCtx.Config,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("create app db: %v", err)
+	}
+	env.DeferErr("close app db", appdb.Close)
+	blockdb := dbm.NewMemDB()
+	env.DeferErr("close block db", blockdb.Close)
+	txdb := cometdb.NewMemDB()
+	env.DeferErr("close tx db", txdb.Close)
+	mempooldb := dbm.NewMemDB()
+	env.DeferErr("close mempool db", mempooldb.Close)
+
+	monomerGenesisPath := viper.GetString(monomerGenesisPathFlag)
+	svrCtx.Logger.Info("Loading Monomer genesis from", "path", monomerGenesisPath)
+
+	appGenesis, err := genutiltypes.AppGenesisFromFile(monomerGenesisPath)
+	if err != nil {
+		return nil, fmt.Errorf("load application genesis file: %v", err)
+	}
+
+	genChainID, err := strconv.ParseUint(appGenesis.ChainID, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("parse chain ID: %v", err)
+	}
+
+	var appState map[string]json.RawMessage
+	if err := json.Unmarshal(appGenesis.AppState, &appState); err != nil {
+		return nil, fmt.Errorf("unmarshal app state: %v", err)
+	}
+
+	n := node.New(
+		wrappedApp,
+		&genesis.Genesis{
+			ChainID:  monomer.ChainID(genChainID),
+			AppState: appState,
+		},
+		engineWS,
+		cometListener,
+		blockdb,
+		mempooldb,
+		txdb,
+		&node.SelectiveListener{
+			OnEngineHTTPServeErrCb: func(err error) {
+				svrCtx.Logger.Error("[Engine HTTP Server]", "error", err)
+			},
+			OnEngineWebsocketServeErrCb: func(err error) {
+				svrCtx.Logger.Error("[Engine Websocket]", "error", err)
+			},
+			OnCometServeErrCb: func(err error) {
+				svrCtx.Logger.Error("[CometBFT]", "error", err)
+			},
+		},
+	)
+	svrCtx.Logger.Info("Spinning up Monomer node")
+
+	nodeCtx, nodeCtxCancel := context.WithCancel(context.Background())
+	env.Defer(nodeCtxCancel)
```

Comment:
Let's move this context to the top-level `StartCommandHandler` and pass it down to all functions that require a context. This will also help resolve the context issue mentioned in a comment above.